Pocket/pocket-ios
# feat(search): entry points

## Summary
* Entry points for Search

## References 
* [IN-887](https://getpocket.atlassian.net/jira/software/projects/IN/boards/80?selectedIssue=IN-887)
* [IN-956](https://getpocket.atlassian.net/jira/software/projects/IN/boards/80?selectedIssue=IN-956)

## Implementation Details
* added navigationItem search bar to SavesContainerViewController
* handled navigation via CompactSaves and RegularSaves Coordinators

## Test Steps
* on phone & tablet:
  * Go to Saves
    * swipe down to reveal search bar
    * click the search bar
    * confirm that the scope is set to Saves
    * navigate back to Saves
    * tap the Search Carousel item
    * confirm that the scope is set to Saves
  * Go to Archives
    * swipe down to reveal search bar (if not already visible)
    * click the search bar
    * confirm that the scope is set to Archives
    * navigate back to Archives
    * tap the Search Carousel item
    * confirm that that scope is set to Archives 

## PR Checklist:

## Screenshots
tested both entry from carousel and search bar
https://user-images.githubusercontent.com/111293209/202278084-a580dee7-1846-4647-8081-02ba4b58c641.mp4
